src/tools/data/datagen/testall.C

```diff
@@ -540,6 +543,25 @@ build_rect2d(DBfile * dbfile, int size)
     fill_rect2d_mat (x, y, matlist, nx, ny, mix_next, mix_mat, mix_zone,
                      mix_vf, &mixlen, 18, 0.1);
 
+    //
+    // Calculate dmix.
+    //
+    for (i = 0; i < nx * ny; i++)
+    {
+        if (matlist[i] > 0)
+        {
+            d[i] = matlist[i] * 0.2;
+        }
+        else
+        {
+            int m1 = - matlist[i] - 1;
+            int m2 = mix_next[m1] - 1;
+            mix_dmix[m1] = mix_mat[m1] * 0.2;
+            mix_dmix[m2] = mix_mat[m2] * 0.2;
+            d[i] = mix_dmix[m1] * mix_vf[m1] + mix_dmix[m2] * mix_vf[m2];
+        }
+    }
+
     if (mixlen > 40 * ny)
     {
         printf ("mixlen = %d\n", mixlen);
```

Comment:
Lets improve this error message here by saying something like `printf(stderr, "ERROR: Hard-coded mixlen exceeded at line %d\n", __LINE__);`. Would any developer understand what seeing `mixlen = 1200` on the erminal would even suggest there was an error?

Reply:
FYI...it turns out there are two instances of this error message in this file that could stand the same treatment

Reply:
I fixed that error message and improved the others. I actually removed one of the error messages, since it could never be executed.